src/openrct2/object/TerrainSurfaceObject.cpp

```diff
@@ -136,24 +138,43 @@ void TerrainSurfaceObject::ReadJson(IReadObjectContext* context, json_t& root)
     PopulateTablesFromJson(context, root);
 }
 
-uint32_t TerrainSurfaceObject::GetImageId(
-    const CoordsXY& position, int32_t length, int32_t rotation, int32_t offset, bool grid, bool underground) const
+ImageId TerrainSurfaceObject::GetImageId(
+    const CoordsXY& position, uint8_t length, uint8_t rotation, uint8_t offset, bool grid, bool underground) const
 {
-    uint32_t result = (underground ? DefaultUndergroundEntry : (grid ? DefaultGridEntry : DefaultEntry));
+    uint32_t result = DefaultEntry;
+    const std::vector<SpecialEntry>* entries = &SpecialEntries;
```

Comment:
I suggest to use std::span, this should work if I'm not mistaken.
```suggestion
    auto entries = std::span<const SpecialEntry>{ SpecialEntries };
```